wsmaster/che-core-api-workspace-activity/src/main/java/org/eclipse/che/api/workspace/activity/InmemoryWorkspaceActivityDao.java

```diff
@@ -139,6 +140,15 @@ public void removeActivity(String workspaceId) throws ServerException {
     workspaceActivities.remove(workspaceId);
   }
 
+  @Override
+  public void createActivity(WorkspaceActivity activity) throws ConflictException {
+    if (workspaceActivities.containsKey(activity.getWorkspaceId())) {
+      throw new ConflictException("Already exists.");
```

Comment:
Could you please add more details in error message, like
`Workspace activity record for workspace with id `%s` already exists`

Reply:
As far as I looked `InmemoryWorkspaceActivityDao` is actually not used. I just didn't want to remove it as part of this PR ;)